FWCore/ParameterSet/interface/ParameterSetDescriptionFillerBase.h

```diff
@@ -78,12 +78,13 @@ namespace edm {
 
     // ---------- member functions ---------------------------
 
-  protected:
-    static const std::string kEmpty;
-    static const std::string kBaseForService;
     static const std::string kBaseForESSource;
     static const std::string kBaseForESProducer;
     static const std::string kBaseForEDLooper;
```

Comment:
Maybe there is a better way than making these public? Although maybe it isn't too bad either?

(in the Event system the EDModule base classes define the corresponding strings)

Reply:
Is it possible to use constexpr std::string_view for this to avoid allocation when the shared library is loaded? [Although many implementations of std::string have a small string optimization]

Reply:
Sounds plausible, but I'd leave that to a subsequent PR because of being somewhat orthogonal.